azurerm/resource_arm_private_dns_a_record.go

```diff
@@ -0,0 +1,188 @@
+package azurerm
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/Azure/azure-sdk-for-go/services/privatedns/mgmt/2018-09-01/privatedns"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmPrivateDnsARecord() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmPrivateDnsARecordCreateUpdate,
+		Read:   resourceArmPrivateDnsARecordRead,
+		Update: resourceArmPrivateDnsARecordCreateUpdate,
+		Delete: resourceArmPrivateDnsARecordDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
```

Comment:
> It seems everything than lower case characters in the resource group name trigger a resource recreation. I could use some help where to look to fix that.

Unfortunately this would be a bug in the Azure API - whilst officially the Azure API Specification says that the Resource Group names are case insensitive; some API's require that they're case sensitive; as such Terraform treats them as case-sensitive by default. In this instance I'd recommend [opening a bug on the Rest API Specs Repository](https://github.com/Azure/azure-rest-api-specs) where this can be tracked; and adding a TODO with a link to that (see below for an example) - and we can work around this in the interim using:

```suggestion
			// TODO: make this case sensitive once the API's fixed [link]
			"resource_group_name": azure.SchemaResourceGroupNameDiffSuppress(),
```